docs/runtime-contract.md

```diff
@@ -80,35 +79,23 @@ serverless platform SHOULD observe the following properties:
   [this issue](https://github.com/knative/serving/issues/848) for reasons an
   operator might want to de-allocate CPU between requests).
 
-### State
-
-The general OCI state may not be available for introspection within the
-container, and may only be visible to the system operator or platform provider.
-In a highly-shared environment, containers may experience the following:
-
-- Containers with `status` of `stopped` MAY be immediately reclaimed by the
-  system.
-- The container process MAY be started as pid 0, through the use of PID
-  namespaces or other processes.
-
 ### Lifecycle
 
 - The container MAY be killed when the container is inactive. Serverless
   computation relies on inbound requests to determine container activeness. The
-  container is sent a `SIGTERM` signal when it is killed via the
-  [OCI specification's `kill`](https://github.com/opencontainers/runtime-spec/blob/master/runtime.md#kill)
-  command to allow for a graceful shutdown of existing resources and
-  connections. If the container has not shut down after a defined grace period,
-  the container is forcibly killed via a `SIGKILL` signal.
-- The environment MAY restrict the use of `prestart`, `poststart`, and
-  `poststop` hooks to platform operators rather than developers. All of these
-  hooks are defined in the context of the runtime namespace, rather than the
-  container namespace, and may expose system-level information (and are
+  container MUST be sent a `SIGTERM` signal when it is killed to allow for a
+  graceful shutdown of existing resources and connections. If the container has not
+  shut down after a defined grace period, the container MUST be forcibly killed via
+  a `SIGKILL` signal.
+- The environment
+  [SHOULD](https://github.com/knative/serving/blob/master/test/conformance/container_test.go)
+  restrict the use of `prestart`, `poststart`, and `poststop` hooks to platform
+  operators rather than developers. All of these hooks are defined in the context
+  of the runtime namespace, rather than the
+  container namespace, and MAY expose system-level information (and are
   non-portable).
-- Failures of the developer-specified process MUST be logged to a
```

Comment:
I'm assuming this was removed because it's not currently testable?

Can we leave a comment suggesting this for future inclusion once we have appropriate log infrastructure in place? I think it would be testable if we had a standard log-tail interface:

* start tailing
* Deploy container which runs `echo "Not gonna happen" 1>&2; exit 1`
* Check for string in tail results.